releasenotes/notes/vf2layout-preset-passmanager-db46513a24e79aa9.yaml

```diff
@@ -0,0 +1,22 @@
+---
+upgrade:
+  - |
+    The preset passmanagers in :mod:`qiskit.transpiler.preset_passmanagers`
+    for all optimization levels have been changed to run the
+    :class:`~qiskit.transpiler.passes.VF2Layout` by default prior to the layout
+    pass. The :class:`~qiskit.transpiler.passes.VF2Layout` pass will quickly
+    check if a perfect layout can be found and superscedes what was previously
+    done for optimization levels 2 and 3 which were using a combination of
+    :class:`~qiskit.transpiler.passes.TrivialLayout` and
+    :class:`~qiskit.transpiler.passes.CSPLayout` to try and find a perfect
+    layout. However, because :class:`~qiskit.transpiler.passes.VF2Layout` is
+    fast even in the failure mode it has been added to level level 1 as
+    well to improve the quality of the results. This will result in potentially
+    different behavior when :func:`~qiskit.compiler.transpile` is called by
+    default as it removes a default path for all optimization levels >=1
+    of using a trivial layout (where circuit.qubits[0] is mapped to physical
+    qubit 0, circuit.qubits[1] is mapped to physical qubit 1, etc) assuming
+    the trivial layout is perfect. If your use case was dependent on the
+    trivial layout you can explictly request it when transpiling by specifying
+    ``layout_method="trivial"`` when calling :func:`~qiskit.compiler.transpile`.
```

Comment:
Maybe I slightly misunderstood the code, but if you specify `layout_method="trivial"`, won't it still run VF2, and only if that fails to find an exact match will it run `"trivial"`?  So setting `layout_method` doesn't guarantee that you'll get what you asked for.

Reply:
If you explicitly set a layout method it skips vf2 that's what this logic is about: https://github.com/Qiskit/qiskit-terra/pull/7213/files#diff-c77fdb2238b27aaeeefbed4c0f92992b6602e4633f0c8960e2f509b444e5396eR128 (this is a carryover from how `layout_method` worked with csp before)

Reply:
Ah yep, sorry - missed that line.